cuqi/distribution/_joint_distribution.py

```diff
@@ -294,7 +298,90 @@ def logpdf(self, stacked_input):
         return self.logd(stacked_input)
     
     def _sample(self, Ns=1):
-        raise TypeError("StackedJointDistribution does not support sampling.")
+        raise TypeError(f"{self.__class__.__name__} does not support sampling.")
 
     def __repr__(self):
         return "_Stacked"+super().__repr__()
+
+
+class SingleVariablePosterior(JointDistribution, Distribution):
```

Comment:
In our discussion today we agreed to rename back to `MultipleLikelihoodPosterior`, which makes it clearer what this class is, that it is a more general form of posterior.